src/ucp/core/ucp_request.h

```diff
@@ -98,7 +99,11 @@ struct ucp_request {
             ucp_send_callback_t   cb;       /* Completion callback */
 
             union {
-                ucp_tag_t         tag;      /* Tagged send */
+                struct {
+                    ucp_tag_t     tag;      /* Tagged send */
+                    uint64_t      message_id; /* message ID used in AM */
+                };
```

Comment:
Why is this anonymous? Looks like it's applicable only for tag/am, right? BTW, do you plan to expand this on stream?

Reply:
it is anonymous to avoid merge conflicts. it will be wrapped into regular structure after AM multi-lane completes

Reply:
Ok, though i don't understand what's a problem.. what about stream/am?